pageserver/src/tenant.rs

```diff
@@ -639,26 +642,59 @@ impl Tenant {
             .as_ref()
             .ok_or_else(|| anyhow::anyhow!("cannot attach without remote storage"))?;
 
-        let remote_timelines = remote_timeline_client::list_remote_timelines(
+        let remote_timeline_ids = remote_timeline_client::list_remote_timelines(
             remote_storage,
             self.conf,
             self.tenant_id,
         )
         .await?;
 
-        info!("found {} timelines", remote_timelines.len());
+        info!("found {} timelines", remote_timeline_ids.len());
 
-        let mut timeline_ancestors: HashMap<TimelineId, TimelineMetadata> = HashMap::new();
-        let mut index_parts: HashMap<TimelineId, IndexPart> = HashMap::new();
-        for (timeline_id, index_part) in remote_timelines {
-            let remote_metadata = index_part.parse_metadata().with_context(|| {
-                format!(
-                    "Failed to parse metadata file from remote storage for tenant {} timeline {}",
-                    self.tenant_id, timeline_id
+        // Download & parse index parts
+        let mut part_downloads = FuturesUnordered::new();
```

Comment:
NIT

> FuturesUnordered

Before I learned about [`JoinSet`](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html), I've also used it, but it requires an extra package, `futures`, while `JoinSet` is in `tokio` which we're using heavily anyway.

JoinSet's signatures seem to be slightly simply too, so we might consider using that in the end.

Reply:
Cool, will switch to it.